charts/humio-fluentbit/templates/fluent-bit-ds.yaml

```diff
@@ -49,6 +49,8 @@ spec:
         - name: wait-for-shared-tokens
           image: {{ default "humio/strix" .Values.strixImage }}
           imagePullPolicy: {{ default "Always" .Values.strixImagePullPolicy }}
+          imagePullSecrets:
+            {{- toYaml . | nindent 12 }}
```

Comment:
This doesn't seem to be completely correct to me. If you compare this with the example above for the "wait-for-humio" init container, it uses a `{{- with .Values.imagePullSecrets }}` part which is what makes it possible to refer to the content list of secrets by using `.` as part of `{{- toYaml . | nindent 12 }}`

Reply:
Actually. Looks like the `imagePullSecrets` is misplaced and instead of being added to containers/initContainers section it should be specified directly on the Pod spec. See an example here: https://kubernetes.io/docs/concepts/containers/images/#referring-to-an-imagepullsecrets-on-a-pod

Perhaps you want to fix that bug as part of this?

Thanks!

Reply:
updated thanks

Reply:
@darren-rose Just making sure, did you see my second comment I posted above? I noticed that this was wrong all along also for the part I referred to. So essentially we'd need to: remove imagePullSecrets from containers/initContainers and add it to the DaemonSet.Spec similar to how we do it for the humio-strix chart (humio-strix uses Deployment object but the idea is the same): https://github.com/humio/humio-helm-charts/blob/master/charts/humio-strix/templates/deployment.yaml#L18-L22

Reply:
nice catch, updated